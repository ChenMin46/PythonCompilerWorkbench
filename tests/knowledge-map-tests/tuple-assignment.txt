a, b = c, d
(a, b) = (c, d)

# multiline
(a, b,
   c) = \
     (c, d)
