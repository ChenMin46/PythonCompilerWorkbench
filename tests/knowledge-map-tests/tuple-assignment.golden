a, b = c, d
(a, b) = (c, d)

# multiline
(a, b,
   c) = \
     (c, d)
===
Module
  body: [
    Assign >> 'a, b = c, d'
      targets: [
        Tuple >> 'a, b'
          elts: [
            Name >> 'a'
              id: 'a'
              ctx: Store
            Name >> 'b'
              id: 'b'
              ctx: Store
          ]
          ctx: Store
      ]
      value: Tuple >> 'c, d'
        elts: [
          Name >> 'c'
            id: 'c'
            ctx: Load
          Name >> 'd'
            id: 'd'
            ctx: Load
        ]
        ctx: Load
    Assign >> '(a, b) = (c, d)'
      targets: [
        Tuple >> '(a, b)'
          elts: [
            Name >> 'a'
              id: 'a'
              ctx: Store
            Name >> 'b'
              id: 'b'
              ctx: Store
          ]
          ctx: Store
      ]
      value: Tuple >> '(c, d)'
        elts: [
          Name >> 'c'
            id: 'c'
            ctx: Load
          Name >> 'd'
            id: 'd'
            ctx: Load
        ]
        ctx: Load
    Assign ML>> '(a, b,\n   c) = \\\n     (c, d)'
      targets: [
        Tuple ML>> '(a, b,\n   c)'
          elts: [
            Name >> 'a'
              id: 'a'
              ctx: Store
            Name >> 'b'
              id: 'b'
              ctx: Store
            Name >> 'c'
              id: 'c'
              ctx: Store
          ]
          ctx: Store
      ]
      value: Tuple >> '(c, d)'
        elts: [
          Name >> 'c'
            id: 'c'
            ctx: Load
          Name >> 'd'
            id: 'd'
            ctx: Load
        ]
        ctx: Load
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    "a"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ", ", 
                {
                  "contents": [
                    "b"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }, 
            " = ", 
            {
              "contents": [
                {
                  "contents": [
                    "c"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ", ", 
                {
                  "contents": [
                    "d"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "(", 
                {
                  "contents": [
                    "a"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ", ", 
                {
                  "contents": [
                    "b"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }, 
            " = ", 
            {
              "contents": [
                "(", 
                {
                  "contents": [
                    "c"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ", ", 
                {
                  "contents": [
                    "d"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        "\n\n# multiline\n", 
        {
          "contents": [
            {
              "contents": [
                "(", 
                {
                  "contents": [
                    "a"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ", ", 
                {
                  "contents": [
                    "b"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ",\n   ", 
                {
                  "contents": [
                    "c"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }, 
            " = \\\n     ", 
            {
              "contents": [
                "(", 
                {
                  "contents": [
                    "c"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ", ", 
                {
                  "contents": [
                    "d"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
